level1c4pps/isccpng2pps_lib.py

```diff
@@ -0,0 +1,268 @@
+    '55_day': [0.00444712, 0.0162967, 0.00000, -2.83943, -0.417072, -0.181784,
+               -0.486767, 0.304194, 0.134653, 0.0706959, 12.3763],
+    '55_all': [0.00000, 0.00000, 0.00000, -1.94343, -0.302762, -0.259928,
+               -0.471424, 0.131567, 0.104622, 0.0658078, 9.71430]
+}
+satellite_names = {270: "GOES-16",  # ABI
+                   271: "GOES-17",  # ABI
+                   173: "Himawari-8",  # AHI
+                   55: "Meteosat-8",  # MSG1 SEVIRI
+                   70: "Meteosat-11"}
+
+
+def get_encoding_isccpng(scene):
+    """Get netcdf encoding for all datasets."""
+    return get_encoding(scene,
+                        BANDNAMES,
+                        PPS_TAGNAMES,
+                        chunks=None)
+
+
+def set_header_and_band_attrs(scene, orbit_n=00000):
+    """Set and delete some attributes."""
+    nimg = 0  # name of first dataset is image0
+    # Set some header attributes:
+    irch = scene['temp_11_00um']
+    irch.attrs['instrument'] = "seviri"
+    scene.attrs['source'] = "isccpng2pps.py"
+    scene.attrs['platform_name'] = "meteosat11"
+    nimg = set_header_and_band_attrs_defaults(scene, BANDNAMES, PPS_TAGNAMES, REFL_BANDS, irch, orbit_n=orbit_n)
+    for band in REFL_BANDS:
+        if band not in scene:
+            continue
+        scene[band].attrs['sun_zenith_angle_correction_applied'] = 'False'
+    return nimg
+
+
+def homogenize_channel(scene, wmo_id, illum, band, sol_zen):
+    """Homogenize one satellite for one band."""
+    data = scene[band]
+    dwmo_id = scene["wmo_id"].values
+    ch_index = coef_slope_chan.index(band)
+    k = coef_slope_list[str(wmo_id) + illum][ch_index]
+    c = coef_offset_list[str(wmo_id) + illum][ch_index]
+    logger.info(
+        f"Homogenizing channel {band} for {satellite_names[wmo_id]} for illum:{illum}"
+        f" with respect to SEVIRI on MSG4 using y={k} * x + {c}")
+    dn_discr = 90
+    if (illum == '_day'):
+        update = ((dwmo_id == wmo_id) & (data > 0) & (sol_zen < dn_discr))
+        scene[band].values = np.where(update, scene[band].values * k + c, scene[band].values)
+
+    if (illum == '_nig'):
+        update = ((dwmo_id == wmo_id) & (data > 0) & (sol_zen >= dn_discr))
+        scene[band].values = np.where(update, scene[band].values * k + c, scene[band].values)
+
+
+def homogenize(scene):
+    """Homogenize data to Meteosat-11."""
+    sol_zen = scene["sunzenith"]
+    for band in BANDNAMES:
+        for wmo_id in [270, 271, 173, 55]:
+            homogenize_channel(scene, wmo_id, '_day', band, sol_zen)
+            homogenize_channel(scene, wmo_id, '_nig', band, sol_zen)
+
+def recalibrate_meteosat(scene):
+    """Nominal calibration is applied, redo with meirnik calibration."""
+    from satpy.readers.seviri_base import MeirinkCalibrationHandler
+    start_time = dt64_to_datetime(scene["refl_00_65um"].attrs["start_time"])
+    channel_name = {"refl_00_65um": "VIS006",
```

Comment:
I prefer to define constants like this and the ones below in the top of the module.